### PR TITLE
zos: update child-tree retrieval and applicable tests

### PR DESCRIPTION
**Pull request by gabylb**
#### Checklist
- [x] npm install && npm test passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

#### Description of change
- the current algorithm for retrieving a given process's children tree assumes ps lists a parent process before its child; however, on z/OS this is not always the case; this update traverses the ps list recursively, and was applied only when running on z/OS
- one of the tests reads from /proc, which doesn't exist on z/OS and has been disabled

### CODE REVIEW

**Comment by gabylb** [comment]
Hello @remy, please review/merge when you have a chance.

File: lib/tree.js [hunk 1]
```diff
@@ -1,10 +1,51 @@
 const spawn = require('child_process').spawn;
 
-module.exports = function(rootPid, callback) {
+function find_kids_os390(rootPid, callback) {
```
Comment (remy) [inline lib/tree.js]: Please rename to remove "kids" from this code. My own code policy after losing my own child is not to refer to children, kids, etc when meaning processes (this does appear in old code, but I'm making sure new code doesn't come in with it).

You'll also need to rebase your code from the latest PR - which isn't a huge change.

**Review by remy — CHANGES_REQUESTED** [review]

**Comment by remy** [comment]
This should also ship with a test with a fixture that has the output from this zos `ps` report.

**Comment by gabylb** [comment]
There was an issue with the rebase that I couldn't resolve, I'll open a new PR instead.

**Comment by gabylb** [comment]
New PR to replace this: https://github.com/remy/pstree/pull/17

**Pull request closed without merging**